ArmPlatformPkg/Sec/Sec.c

```diff
@@ -86,18 +86,27 @@ STATIC CONST EFI_PEI_PPI_DESCRIPTOR  gCommonPpiTable[] = {
 
   @param[out]   PpiListSize   Size of the PPI list in bytes
   @param[out]   PpiList       Pointer to the constructed PPI list
+  @param[in]    TransferListBase  Pointer to the Transfer List base
 **/
 STATIC
 VOID
 CreatePpiList (
   OUT UINTN                   *PpiListSize,
-  OUT EFI_PEI_PPI_DESCRIPTOR  **PpiList
+  OUT EFI_PEI_PPI_DESCRIPTOR  **PpiList,
+  IN  VOID                    *TransferListBase
   )
 {
   EFI_PEI_PPI_DESCRIPTOR  *PlatformPpiList;
   UINTN                   PlatformPpiListSize;
   UINTN                   ListBase;
-  EFI_PEI_PPI_DESCRIPTOR  *LastPpi;
+
+  // The Transfer List PPI is used to pass the Transfer List to the PEI Core
+  // and goes at the end of the PPI list.
+  EFI_PEI_PPI_DESCRIPTOR  mTransferListBase = {

```

Comment:
Please drop this definition entirely (note btw that the `m` prefix is not appropriate for stack variable).

Instead, use `LastPpi` - see below